src/MomentoCacheClient.php

```diff
@@ -2367,53 +2368,24 @@ public function zunionstore(string $dst, array $keys, ?array $weights = null, ?s
             return false;
         }
 
-        // Perform asynchronous fetches from each sorted set
-        $futures = [];
-        foreach ($keys as $key) {
-            $futures[] = $this->client->sortedSetFetchByScoreAsync($this->cacheName, $key);
-        }
-
-        // Wait for all the asynchronous fetches to complete
-        $results = [];
-        foreach ($futures as $future) {
-            $response = $future->wait();
-            if ($response->asHit()) {
-                $results[] = $response->asHit()->valuesArray();
-            } elseif ($response->asMiss()) {
-                $results[] = [];
-            } elseif ($response->asError()) {
-                return MomentoToPhpRedisExceptionMapper::mapExceptionElseReturnFalse($response);
-            } else {
-                return false;
-            }
-        }
-
-        // Aggregate the fetched results
-        $union = [];
-        foreach ($results as $index => $set) {
-            $weight = $weights[$index];
-            foreach ($set as $member => $score) {
-                if (!isset($union[$member])) {
-                    $union[$member] = $weight * $score;
-                } else {
-                    $weightedScore = $weight * $score;
-                    if ($aggregate === ZunionstoreOptionsHelper::AGGREGATE_SUM) {
-                        $union[$member] += $weightedScore;
-                    } elseif ($aggregate === ZunionstoreOptionsHelper::AGGREGATE_MIN) {
-                        $union[$member] = min($union[$member], $weightedScore);
-                    } elseif ($aggregate === ZunionstoreOptionsHelper::AGGREGATE_MAX) {
-                        $union[$member] = max($union[$member], $weightedScore);
-                    }
-                }
-            }
+        $aggregateFunctionMap = [
+            ZunionstoreOptionsHelper::AGGREGATE_SUM => SortedSetUnionStoreAggregateFunction::SUM,
+            ZunionstoreOptionsHelper::AGGREGATE_MIN => SortedSetUnionStoreAggregateFunction::MIN,
+            ZunionstoreOptionsHelper::AGGREGATE_MAX => SortedSetUnionStoreAggregateFunction::MAX,
+        ];
```

Comment:
Could move this over to the ZunionstoreOptionsHelper to just convert straight to the Momento type